python/taichi/lang/kernel_arguments.py

```diff
@@ -82,14 +82,10 @@ def get_type_for_kernel_args(dtype, name):
 
 
 def decl_matrix_arg(matrixtype, name):
-    if isinstance(matrixtype, VectorType):
-        return make_matrix([decl_scalar_arg(matrixtype.dtype, f"{name}_{i}") for i in range(matrixtype.n)])
-    return make_matrix(
-        [
-            [decl_scalar_arg(matrixtype.dtype, f"{name}_{i}_{j}") for i in range(matrixtype.m)]
-            for j in range(matrixtype.n)
-        ]
-    )
+    arg_type = get_type_for_kernel_args(matrixtype, name)
+    arg_id = impl.get_runtime().compiling_callable.insert_scalar_param(arg_type, name)
```

Comment:
nit: are we going to rename `insert_scalar_param` in the future PRs?

Reply:
Maybe we can merge `insert_scalar_param` and `insert_pointer_param` to a `insert_param` with an option `is_ptr`